editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -581,429 +572,492 @@ async fn create_compound_image(
+            .set_json(json!({ "locked": false }))
+            .to_request();
+
+        let response = call_service(&app.service, request).await;
+        assert_eq!(response.status(), StatusCode::NO_CONTENT);
+
+        let fast_rolling_stock: RollingStockModel =
+            RollingStockModel::retrieve(db_pool.get_ok().deref_mut(), locked_fast_rolling_stock.id)
+                .await
+                .expect("Failed to retrieve rolling stock")
+                .expect("Rolling stock not found");
+
+        assert!(!fast_rolling_stock.locked);
+    }
+
+    #[rstest]
+    async fn get_power_restrictions_list() {
         // GIVEN
-        let other_rs_name = "other_rolling_stock_update_rolling_stock_failure_name_already_used";
-        let app = create_test_service().await;
-        let fast_rolling_stock = named_fast_rolling_stock(
-            "fast_rolling_stock_update_rolling_stock_failure_name_already_used",
-            db_pool.clone(),
-        )
-        .await;
-        let _other_rs = named_other_rolling_stock(other_rs_name, db_pool).await;
+        let app = TestAppBuilder::default_app();
+        let db_pool = app.db_pool();
 
-        let rolling_stock_id = fast_rolling_stock.id();
+        let rs_name = "fast_rolling_stock_name";
+        let fast_rolling_stock =
+            create_fast_rolling_stock(db_pool.get_ok().deref_mut(), rs_name).await;
+        let power_restrictions = fast_rolling_stock.power_restrictions.clone();
 
-        let rolling_stock_form = get_other_rolling_stock_form(other_rs_name);
+        let request = TestRequest::get()
+            .uri("/rolling_stock/power_restrictions")
+            .to_request();
 
         // WHEN
-        let response = call_service(
-            &app,
-            TestRequest::patch()
-                .uri(format!("/rolling_stock/{}", rolling_stock_id).as_str())
-                .set_json(rolling_stock_form)
-                .to_request(),
-        )
-        .await;
+        let response = call_service(&app.service, request).await;
 
         // THEN
-        assert_eq!(response.status(), StatusCode::BAD_REQUEST);
-        assert_response_error_type_match!(
-            response,
-            RollingStockError::NameAlreadyUsed {
-                name: String::from(other_rs_name),
-            }
-        );
+        let power_restrictions = serde_json::to_string(&power_restrictions)
+            .expect("Failed to convert power_restrictions to string");
+        assert!(power_restrictions.contains(&"C2".to_string()));
+        assert!(power_restrictions.contains(&"C5".to_string()));
+        let response_body: Vec<String> = read_body_json(response).await;
+        assert!(response_body.contains(&"C2".to_string()));
+        assert!(response_body.contains(&"C5".to_string()));
     }
 
     #[rstest]
-    async fn update_locked_successfully(db_pool: Arc<DbConnectionPool>) {
+    #[ignore] // TODO test for rolling stock with DBConnectionPoolV2
```

Comment:
Is it planned to be follow-up in a future PR, or is it something you forgot to address on the current PR?
(note: there is a few others TODO below too).

Reply:
It using the train schedule V1, with @leovalais we decided to wait the removal of train schedule V1 to finish